### PR TITLE
28 entity fields part 1

### LINKED ISSUE #28: Add fields to entity (I)
Add the following fields to the base Entity model.

- Type (Company or Person)
- Tax No
- Tax State
  - List of states
- Tax Code
- Federal Tax ID
- Print 1099 (Yes, No)
- 1099 Name
- 1099 Form
  - Crop Insurance Proceeds
  - Direct Sales
  - Excess Golden Parachute Payments
  - Federal Income Tax Withheld
  - Fishing Boat Proceeds
  - Gross Proceeds Paid to an Attorney
  - Medical and Health Care Payments
  - Non-employee Compensation
  - Other Income
  - Rents
  - Royalties
  - Substitute Payments in Lieu of Dividends or Interest
- 1099 Type
  - None
  - 1099-Misc
  - 1099-Int
  - 1099-B
- W9 Signed (Date)
- Withholding (Yes, No)

### PR DESCRIPTION
**Pull request by josephbridgwaterrowe**
Closes #28

### CODE REVIEW

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco please review.

File: app/controllers/entities_controller.rb [hunk 1]
```diff
@@ -58,9 +57,18 @@ def entity_hash
     @entity_entry
       .to_h
       .slice(:display_name,
-             :parent_entity_id,
+             :entity_type,
```
Comment (michaelachrisco) [inline app/controllers/entities_controller.rb]: Is the `Entity No` the entity id in rails or is this going to be a different field altogether?

Comment (josephbridgwaterrowe) [inline app/controllers/entities_controller.rb]: I don't see an Entity Number, the `reference` would be the equivalent of an Entity, Customer, Vendor Number.

File: app/controllers/entities_controller.rb [hunk 2]
```diff
@@ -58,9 +57,18 @@ def entity_hash
...
+             :federal_tax_id,
+             :is_active,
```
Comment (michaelachrisco) [inline app/controllers/entities_controller.rb]: Are you going to add the other fields in the Payment Control?

**Comment by michaelachrisco** [comment]
Looks good. Other than my one question, should be good to merge.

File: app/controllers/entities_controller.rb (hunk 2 above)
Comment (josephbridgwaterrowe) [inline app/controllers/entities_controller.rb]: Such as?

Comment (michaelachrisco) [inline app/controllers/entities_controller.rb]:
```
Always Discount
EFT
Hold (Is this is_withholding?)
```

Comment (josephbridgwaterrowe) [inline app/controllers/entities_controller.rb]: Where do these come from, am I missing something? :smile:

Comment (michaelachrisco) [inline app/controllers/entities_controller.rb]: screen shot 2015-06-23 at 2 13 38 pm
It is within #28 as the checkboxes. You could roll these into a `entity.payment_control.status` or as a couple more fields added to the model.

**Comment by josephbridgwaterrowe** [comment]
Thanks, no they aren't in the current requirements.

**Comment by michaelachrisco** [comment]
Ok rebase and Ill merge.

**Comment by josephbridgwaterrowe** [comment]
:shipit:

**Pull request merged**